### PR TITLE
[luxon-business-days] New type definitions

### PR DESCRIPTION
**Pull request by bcbailey**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:

- [X] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [X] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [X] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [X] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [X] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/luxon-business-days/luxon-business-days-tests.ts
1. types/luxon-business-days/index.d.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against e221606cd2876a3b0d97713026b6a4eec29c3018
</p>

File: types/luxon-business-days/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+export type HolidayHelper = (...args: any[]) => any;
+export interface HolidayHelpersDict {
+    [key: string]: HolidayHelper;
+}
+
+declare module "luxon" {
+    interface DateTime<IsValid extends boolean> {
+        holidayHelpers: HolidayHelper[] | undefined;
+        holidayMatchers: HolidayMatcher[] | undefined;
```
Comment (peterblazejewicz) [inline types/luxon-business-days/index.d.ts:19]:
```suggestion
        holidayHelpers?: HolidayHelper[] | undefined;
        holidayMatchers?: HolidayMatcher[] | undefined;
```
not sure what was the intent, but these looks like strict optional properties:
https://www.typescriptlang.org/tsconfig/#exactOptionalPropertyTypes

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@bcbailey thanks!

File: types/luxon-business-days/index.d.ts (hunk 1 above)
Comment (bcbailey) [reply types/luxon-business-days/index.d.ts]: Thanks for the review @peterblazejewicz 
I have updated the PR to make them optional. That actually made me catch a mistake I had made (`holidayHelpers` should've been `businessDays`)

**Review by peterblazejewicz — APPROVED** [review]

**Comment by bcbailey** [comment]
Ready to merge

**Pull request merged**